nilearn/decomposition/_base.py

```diff
@@ -417,7 +418,7 @@ def fit(
         ):
             imgs = resolve_globbing(imgs)
 
-        if isinstance(imgs, str) or not hasattr(imgs, "__iter__"):
+        if isinstance(imgs, (str, Path)) or not hasattr(imgs, "__iter__"):
```

Comment:
The docstring tells that `imgs` should be "list of Niimg-like objects". If we are expecting `str` or `Path` as well, it might be good to update docstring accordingly.

Reply:
We don't have typing in nilearn but as far as I understand if we did we would have

```python
niimg_like_object: nibabel.nifti1.Nifti1Image | str | pathlib.Path
```

See : https://nilearn.github.io/stable/manipulating_images/input_output.html#file-names-and-objects-3d-and-4d-images

Reply:
I meant the below part of the code for `fit` function.

```
        Parameters
        ----------
        imgs : list of Niimg-like objects
            See :ref:`extracting_data`.
```
which displays https://nilearn.github.io/stable/modules/generated/nilearn.decomposition.DictLearning.html#nilearn.decomposition.DictLearning.fit .

It could be

```
Parameters
        ----------
        imgs : list of Niimg-like objects, str or Path
            See :ref:`extracting_data`.
```

However I see that, the ref below sends to the link you mentioned.